ros2run/ros2run/command/run.py

```diff
@@ -56,4 +60,5 @@ def main(self, *, parser, args):
             raise RuntimeError(msg)
         if path is None:
             return 'No executable found'
-        return run_executable(path=path, argv=args.argv)
+        prefix=args.prefix[0] if args.prefix is not None else ''
```

Comment:
You could also use the `nargs=*` and require it to be at the end or to be terminated with a `--` (otherwise it will greedily consume the pkg and exec args too).